### PR TITLE
Add a try/catch around registering network callbacks.

### PR DESCRIPTION
**Pull request by jkwatson**
Looks like this is another thing that can fail with the same Android bug (https://issuetracker.google.com/issues/175055271 for reference)

### CODE REVIEW

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java [hunk 1]
```diff
@@ -44,6 +44,14 @@
 
     void startMonitoring(Supplier<NetworkRequest> createNetworkMonitoringRequest, ConnectivityManager connectivityManager) {
         refreshNetworkStatus();
+        try {
+            registerNetworkCallbacks(createNetworkMonitoringRequest, connectivityManager);
+        } catch (Exception e) {
+            //if this fails, we'll go without network change events.
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:50]: Worth logging or making an event or anything?

**Review by breedx-splk — APPROVED** [review]
LGTM, one question about logging, but not a big deal.

File: splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java (hunk 1 above)
Comment (jkwatson) [reply splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:50]: I'm honestly not sure. This is a rare issue, and it happens before the library is initialized, and I've never seen it locally..so...long story short, I don't know what might be useful, and where to put it.

Comment (jkwatson) [reply splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java:50]: I added a warning log. Hopefully if anyone sees it, they will let us know

**Pull request merged**